File: src/main/java/com/movisens/xs/api/models/Message.java
package com.movisens.xs.api.models;

import com.google.gson.annotations.Expose;

import javax.annotation.Generated;

@Generated("org.jsonschema2pojo")
public class Message {

    @Expose
    private long id;
    @Expose
    private String creationDate;
    @Expose
    private String message;
    @Expose
    private boolean messageRead;
    @Expose
    private boolean fromProband;
    @Expose
    private String sendingUserEmail;

    /**
     * @return The id
     */
    public long getId() {
        return id;
    }

    /**
     * @param id The id
     */
    public void setId(long id) {
        this.id = id;
    }

    /**
     * @return The creationDate
     */
    public String getCreationDate() {
        return creationDate;
    }

    /**
     * @param creationDate The creationDate
     */
    public void setCreationDate(String creationDate) {
        this.creationDate = creationDate;
    }

    /**
     * @return The message
     */
    public String getMessage() {
        return message;
    }

    /**
     * @param message The message
     */
    public void setMessage(String message) {
        this.message = message;
    }

    /**
     * @return The messageRead
     */
    public boolean isMessageRead() {
        return messageRead;
    }

    /**
     * @param messageRead The messageRead
     */
    public void setMessageRead(boolean messageRead) {
        this.messageRead = messageRead;
    }

    /**
     * @return The fromProband
     */
    public boolean isFromProband() {
        return fromProband;
    }

    /**
     * @param fromProband The fromProband
     */
    public void setFromProband(boolean fromProband) {
        this.fromProband = fromProband;
    }

    /**
     * @return The sendingUserEmail
     */
    public String getSendingUserEmail() {
        return sendingUserEmail;
    }

    /**
     * @param sendingUserEmail The sendingUserEmail
     */
    public void setSendingUserEmail(String sendingUserEmail) {
        this.sendingUserEmail = sendingUserEmail;
    }

}

File: src/main/java/com/movisens/xs/api/models/Message.kt
package com.movisens.xs.api.models

import com.google.gson.annotations.Expose

class Message {
    @Expose
    var id: Long = 0

    @Expose
    var creationDate: String? = null

    @Expose
    var message: String? = null

    @Expose
    var isMessageRead = false

    @Expose
    var isFromProband = false

    @Expose
    var sendingUserEmail: String? = null
    override fun toString(): String {
        return "Message{" +
                "id=" + id +
                ", creationDate='" + creationDate + '\'' +
                ", message='" + message + '\'' +
                ", messageRead=" + isMessageRead +
                ", fromProband=" + isFromProband +
                ", sendingUserEmail='" + sendingUserEmail + '\'' +
                '}'
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Message

        if (id != other.id) return false
        if (creationDate != other.creationDate) return false
        if (message != other.message) return false
        if (isMessageRead != other.isMessageRead) return false
        if (isFromProband != other.isFromProband) return false
        if (sendingUserEmail != other.sendingUserEmail) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + (creationDate?.hashCode() ?: 0)
        result = 31 * result + (message?.hashCode() ?: 0)
        result = 31 * result + isMessageRead.hashCode()
        result = 31 * result + isFromProband.hashCode()
        result = 31 * result + (sendingUserEmail?.hashCode() ?: 0)
        return result
    }
}